controllers/openstackdataplanenode_controller.go

```diff
@@ -19,10 +19,9 @@ package controllers
 import (
 	"context"
 	"fmt"
-	"strconv"
 
 	"github.com/go-logr/logr"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
```

Comment:
i cant comment on this directly later because github suck for code review
but later we call "by the way i cant comment on this directly because github suck for code review
but later we call "invData, err := yaml.Marshal(inventory)"

if instead we add a Marshal function to the inventory object we can just do"

if instead we add a Marshal function to the inventory object we can just do

invData, err := inventory.Marshal()
and we can avoid importing gopkg.in/yaml.v3 in this file.

in other word i think we should make the Inventory struct be able to Marshal and UnMarshal itself and entirely hide the use of the yaml module in the inventory.go file.